flask/app.py

```diff
@@ -790,25 +785,34 @@ def make_shell_context(self):
             rv.update(processor())
         return rv
 
-    def _reconfigure_for_run_debug(self, debug):
-        """The ``run`` commands will set the application's debug flag. Some
-        application configuration may already be calculated based on the
-        previous debug value. This method will recalculate affected values.
+    #: The environment value.  This is typically set from outside the
+    #: process by setting the `FLASK_ENV` environment variable and can be
+    #: used to quickly switch between different environments like
+    #: `production` and `development`.  If not set this defaults to
+    #: `production`.
+    env = ConfigAttribute('ENV')
 
-        Called by the :func:`flask.cli.run` command or :meth:`Flask.run`
-        method if the debug flag is set explicitly in the call.
+    def _get_debug(self):
+        return self.config['DEBUG']
+    def _set_debug(self, value):
+        self._set_debug_value(value)
 
-        :param debug: the new value of the debug flag
+    #: The debug flag.  Set this to ``True`` to enable debugging of the
+    #: application.  In debug mode the debugger will kick in when an unhandled
+    #: exception occurs and the integrated server will automatically reload
+    #: the application if changes in the code are detected.
+    #:
+    #: This attribute can also be configured from the config with the ``DEBUG``
+    #: configuration key.  Defaults to ``False``.
+    debug = property(_get_debug, _set_debug)
+    del _get_debug, _set_debug
```

Comment:
Use the `property` decorator instead of this pattern.

Reply:
I think it makes sense to change this separately at some point (the other properties also don't use the decorator, so it's consistent for now)

Reply:
Yeah, not too big a deal, but might as well start now.

Reply:
I really don't like the property decorator much. Is there a particular reason you prefer it? It requires reassigning the name over and over and I find it such a weird pattern (and not any shorter)